nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -295,6 +370,11 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
     curvature = 2.0 * carrot_pose.pose.position.y / carrot_dist2;
   }
 
+  double carrot_dist = sqrt(carrot_dist2);
```

Comment:
We use the square intentionally to above the sqrt, can't the `remaining_path_length` be `remaining_path_length*remaining_path_length`, that's much cheaper than a sqrt operation

Reply:
Sqrt was already used for `carrot_dist2` so I felt free to use it (and I really need sqrt):
https://github.com/ros-planning/navigation2/blob/609342fa27016a1677768f93fb3c44dec98b1f55/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp#L328

Reply:
Fair enough - replace the call on that line  then with this variable so we don't do it twice